src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ToolsApiScannerInstaller.java

```diff
@@ -0,0 +1,223 @@
+    private static final String WINDOWS_PLATFORM_PARAMETER_VALUE = "windows";
+
+    public static final String SCAN_CLI_VERSION_FILENAME = "scan-cli-version.txt";
+
+    private final IntLogger logger;
+    private final BlackDuckHttpClient blackDuckHttpClient;
+    private final CleanupZipExpander cleanupZipExpander;
+    private final ScanPathsUtility scanPathsUtility;
+    private final HttpUrl blackDuckServerUrl;
+    private final OperatingSystemType operatingSystemType;
+    private final File installDirectory;
+
+    public ToolsApiScannerInstaller(
+            IntLogger logger,
+            BlackDuckHttpClient blackDuckHttpClient,
+            CleanupZipExpander cleanupZipExpander,
+            ScanPathsUtility scanPathsUtility,
+            HttpUrl blackDuckServerUrl,
+            OperatingSystemType operatingSystemType,
+            File installDirectory
+    ) {
+        if (null == blackDuckServerUrl) {
+            throw new IllegalArgumentException("A Black Duck server url must be provided.");
+        }
+
+        this.logger = logger;
+        this.blackDuckHttpClient = blackDuckHttpClient;
+        this.cleanupZipExpander = cleanupZipExpander;
+        this.scanPathsUtility = scanPathsUtility;
+        this.blackDuckServerUrl = blackDuckServerUrl;
+        this.operatingSystemType = operatingSystemType;
+        this.installDirectory = installDirectory;
+    }
+
+    /**
+     * The Black Duck Signature Scanner (scan-cli) will be downloaded if it has not
+     * previously been downloaded or if it has been updated on the server. The
+     * absolute path to the install location will be returned if it was
+     * downloaded or found successfully, otherwise an Optional.empty will be
+     * returned and the log will contain details concerning the failure.
+     */
+    @Override
+    public File installOrUpdateScanner() throws BlackDuckIntegrationException {
+        if (installDirectory.exists()) {
+            try {
+                ScanPaths scanPaths = scanPathsUtility.searchForScanPaths(installDirectory);
+                if (!scanPaths.isManagedByLibrary()) {
+                    return installDirectory;
+                }
+            } catch (BlackDuckIntegrationException ignored) {
+                // a valid scanPaths could not be found so we will need to attempt an install
+            }
+        }
+
+        File scannerExpansionDirectory = new File(installDirectory, ToolsApiScannerInstaller.BLACK_DUCK_SIGNATURE_SCANNER_INSTALL_DIRECTORY);
+        scannerExpansionDirectory.mkdirs();
+
+        File versionFile = new File(scannerExpansionDirectory, ToolsApiScannerInstaller.SCAN_CLI_VERSION_FILENAME);
+        HttpUrl downloadUrl = getDownloadUrl();
+
+        try {
+            String scannerVersion;
+            if (!versionFile.exists()) {
+                logger.info("No version file exists, assuming this is new installation and the signature scanner should be downloaded.");
```

Comment:
"Signature Scan CLI version file not detected. Default version will be downloaded."